avocado/core/job.py

```diff
@@ -500,9 +500,6 @@ def _run(self):
         if self.status == 'RUNNING':
             self.status = 'PASS'
         # Let's clean up test artifacts
-        if getattr(self.args, 'archive', False):
-            filename = self.logdir + '.zip'
-            archive.create(filename, self.logdir)
```

Comment:
I like you moved this outside the `_run`, which can be potentially not executed on failure